src/Microsoft.Diagnostics.EventFlow.Inputs.NLog/NLogInput.cs

```diff
@@ -0,0 +1,265 @@
+
+        public NLogInput(IHealthReporter healthReporter)
+        {
+            _healthReporter = healthReporter;
+            OptimizeBufferReuse = true;
+            IncludeEventProperties = true;
+            Layout = "${message}";
+
+            RegisterNLogTargetType(healthReporter, GetType());
+
+            _eventFlowSubject = new EventFlowSubject<EventData>();
+        }
+
+        protected override void CloseTarget()
+        {
+            _eventFlowSubject?.Dispose();
+        }
+
+        public IDisposable Subscribe(IObserver<EventData> observer)
+        {
+            return _eventFlowSubject.Subscribe(observer);
+        }
+
+        protected override void Write(LogEventInfo logEvent)
+        {
+            EventData e = ToEventData(logEvent);
+            _eventFlowSubject.OnNext(e);
+        }
+
+        private EventData ToEventData(LogEventInfo logEvent)
+        {
+            EventData eventData = new EventData
+            {
+                ProviderName = nameof(NLogInput),
+                Timestamp = logEvent.TimeStamp.ToUniversalTime(),
+                Level = ToLogLevel[logEvent.Level],
+                Keywords = 0,
+            };
+
+            var payload = eventData.Payload;
+
+            // Prefer the built-in `Message` and `Exception` properties by adding them to the payload
+            // first. If other attached data items have conflicting names, they will be added as
+            // `Message_1` and so-on.
+            if (logEvent.Exception != null)
+            {
+                if (logEvent.Level.Ordinal >= NLog.LogLevel.Error.Ordinal)
+                {
+                    EventMetadata eventMetadata = new EventMetadata(ExceptionData.ExceptionMetadataKind);
+                    eventMetadata.Properties.Add(ExceptionData.ExceptionPropertyMoniker, "Exception");
+                    eventData.SetMetadata(eventMetadata);
+                }
+
+                eventData.AddPayloadProperty("Exception", logEvent.Exception, _healthReporter, nameof(NLogInput));
+            }
+
+            // Inability to render the message, or any other LogEvent property, should not stop us from sending the event down the pipeline
+            try
+            {
+                eventData.AddPayloadProperty("Message", RenderLogEvent(Layout, logEvent), _healthReporter, nameof(NLogInput));
+            }
+            catch (Exception ex)
+            {
+                NLog.Common.InternalLogger.Warn(ex, NLogTargetTypeName + "(Name={0}): Failed to render event message", Name);
+                _healthReporter.ReportWarning($"{nameof(NLogInput)}: event message could not be rendered{Environment.NewLine}{ex.ToString()}");
+                eventData.AddPayloadProperty("Message", logEvent.FormattedMessage, _healthReporter, nameof(NLogInput));
+            }
+
+            // MessageTemplate is always present on Serilog events
```

Comment:
.. but this is NLog input. Is it also the NLog convention to always include the message template?

If so, at leas the comment should be updated.